### PR TITLE
Use child window

### PR DESCRIPTION
**Pull request by AugusteBa**

### CODE REVIEW

**Comment by AugusteBa** [comment]
@oriondean, could you have a look at this PR to see if anything should be changed? I'll update readme tomorrow.

File: demo/src/pages/childWindow/ChildWindow.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    });
+
+    return (
+        <div className={styles.container}>
+            <h1 className={styles.title}>useChildWindow</h1>
+            <div className={styles.description}>
+                This hook returns allows creating and managing child windows.
+            </div>
+            <div className={styles.description}>
+                Try clikcing on a button to see a childWindow launched.
```
Comment (oriondean) [inline demo/src/pages/childWindow/ChildWindow.tsx]: clicking*

File: demo/src/pages/childWindow/ChildWindow.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        parentDocument: document,
+        shouldClosePreviousOnLaunch: true,
+        windowOptions,
+    });
+
+    return (
+        <div className={styles.container}>
+            <h1 className={styles.title}>useChildWindow</h1>
+            <div className={styles.description}>
+                This hook returns allows creating and managing child windows.
```
Comment (oriondean) [inline demo/src/pages/childWindow/ChildWindow.tsx]: Would something like `This hook allows you to create and manage a single child window` be better wording here?

File: demo/src/pages/childWindow/ChildWindow.tsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
+import { useChildWindow } from "openfin-react-hooks";
+import { WindowOption } from "openfin/_v2/api/window/windowOption";
+import React from "react";
+
+import styles from "./ChildWindow.module.css";
+
+const codeExample = `import {useBounds} from "openfin-react-hooks";
```
Comment (oriondean) [inline demo/src/pages/childWindow/ChildWindow.tsx]: This example needs updating

File: index.d.ts [hunk 4]
```diff
@@ -28,6 +29,14 @@ export interface IChannelAction {
   action: Action;
 }
 
+export interface IUseChildWindow {
```
Comment (oriondean) [inline index.d.ts]: IUseChildWindowOptions?

File: package.json [hunk 5]
```diff
@@ -39,6 +39,8 @@
     "react": "^16.8.0"
   },
   "dependencies": {
-    "openfin-layouts": "^1.0.3"
+    "@types/react-dom": "^16.9.1",
```
Comment (oriondean) [inline package.json]: `@types` should be in devDependencies

File: src/useChildWindow.ts [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+    (nodes: HTMLCollectionOf<HTMLStyleElement | HTMLScriptElement>) => {
+      // tslint:disable-next-line: prefer-for-of
+      for (let i = 0; i < nodes.length; i++) {
+        injectNode(nodes[i]);
+      }
+    },
+    [injectNode],
+  );
+
+  const inheritFromParent = useCallback(() => {
```
Comment (oriondean) [inline src/useChildWindow.ts]: Could this behaviour be made optional?

File: src/useChildWindow.ts [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+
+  useEffect(() => {
+    if (childWindow.window) {
+      if (parentDocument) {
+        inheritFromParent();
+      }
+
+      if (cssUrl) {
+        const linkElement = childWindow.window
+        .getWebWindow().document.createElement("link");
```
Comment (oriondean) [inline src/useChildWindow.ts]: indent

**Comment by AugusteBa** [comment]
@oriondean, I addressed your comments, thanks for having a look. Could you look through my changes again, just to make sure that i have got everything right?

File: demo/src/pages/childWindow/interfaces/ILaunchConfig.tsx [hunk 8]
```diff
@@ -0,0 +1,11 @@
+interface ILaunchConfig {
```
Comment (oriondean) [inline demo/src/pages/childWindow/interfaces/ILaunchConfig.tsx]: Could this file just be a definition file instead? e.g. `ILaunchConfig.d.ts` + `export interface ILaunchConfig`

File: src/useChildWindow.ts [hunk 9]
```diff
@@ -0,0 +1,181 @@
...
+    cssUrl,
+    parentDocument,
+    inheritCss,
+    inheritScripts,
+    injectNode,
+  ]);
+
+  useEffect(() => {
+    if (childWindow.state === CHILD_WINDOW_STATE.LAUNCHED) {
+      if (jsx) {
```
Comment (oriondean) [inline src/useChildWindow.ts]: Can combine these two if statements

File: demo/src/pages/home/Home.tsx [hunk 10]
```diff
@@ -88,6 +88,13 @@ const cards: IDemoCard[] = [
       id: "channel-provider",
       name: "useChannels",
     },
+    {
+      description: "Launch and manage child windows",
```
Comment (oriondean) [inline demo/src/pages/home/Home.tsx]: Create and manage a single child window*?

**Pull request merged**